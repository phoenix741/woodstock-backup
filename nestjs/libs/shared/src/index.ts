export * from './authentification';
export * from './backuping';
export * from './commands';
export * from './config';
export * from './core';
export * from './files';
export * from './input-output';
export * from './maintenance';
export * from './manifest';
export * from './pool';
export * from './queue';
export * from './refcnt';
export * from './scanner';
export * from './shared';
export * from './statistics';
export * from './utils';
