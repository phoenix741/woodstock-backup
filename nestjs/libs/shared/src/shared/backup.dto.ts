import { Field, ObjectType } from '@nestjs/graphql';
import { Transform } from 'class-transformer';

@ObjectType()
export class Backup {
  number!: number;
  complete!: boolean;

  startDate!: number;
  endDate?: number;

  fileCount!: number;
  newFileCount!: number;
  existingFileCount!: number;

  @Transform((v) => BigInt(v.value))
  @Field(() => BigInt)
  fileSize!: bigint;
  @Transform((v) => BigInt(v.value))
  @Field(() => BigInt)
  existingFileSize!: bigint;
  @Transform((v) => BigInt(v.value))
  @Field(() => BigInt)
  newFileSize!: bigint;

  @Transform((v) => BigInt(v.value))
  @Field(() => BigInt)
  compressedFileSize!: bigint;
  @Transform((v) => BigInt(v.value))
  @Field(() => BigInt)
  existingCompressedFileSize!: bigint;
  @Transform((v) => BigInt(v.value))
  @Field(() => BigInt)
  newCompressedFileSize!: bigint;

  speed!: number;
}
