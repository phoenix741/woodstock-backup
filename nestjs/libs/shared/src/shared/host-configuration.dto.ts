import { Field, ObjectType } from '@nestjs/graphql';
import { ApiExtraModels, ApiProperty } from '@nestjs/swagger';
import { Exclude } from 'class-transformer';
import { IsNumber, Matches, Max, Min, ValidateNested } from 'class-validator';
import { Schedule } from './scheduler.dto.js';

/**
 * Part of config file.
 *
 * Store information about a share
 */
@ObjectType()
export class BackupTaskShare {
  @ApiProperty({ example: '/home' })
  name!: string;

  @ApiProperty({ example: ['/home/user'] })
  includes?: string[];

  @ApiProperty({ example: ['*.bak'] })
  excludes?: string[];
}

/**
 * Part of config file
 *
 * Store information about a DHCP Address
 */
@ObjectType()
export class DhcpAddress {
  @ApiProperty({ example: '192.168.101' })
  @Matches(/^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){2}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$/)
  address!: string;

  @ApiProperty({ example: 0 })
  @Min(0)
  @Max(255)
  start!: number;

  @ApiProperty({ example: 50 })
  @Min(0)
  @Max(255)
  end!: number;
}

@ObjectType()
export class ExecuteCommandOperation {
  @ApiProperty({ example: '/bin/true' })
  command!: string;
}

@ObjectType()
export class BackupOperation {
  @ValidateNested()
  shares!: Array<BackupTaskShare>;

  @ApiProperty({ example: [] })
  includes?: Array<string>;

  @ApiProperty({ example: [] })
  excludes?: Array<string>;

  @ApiProperty({ example: 1200 })
  @IsNumber()
  timeout?: number;
}

@ApiExtraModels(BackupOperation, ExecuteCommandOperation)
@ObjectType()
export class HostConfigOperation {
  @ApiProperty({ type: [ExecuteCommandOperation] })
  @ValidateNested()
  @Field(() => [ExecuteCommandOperation])
  preCommands?: ExecuteCommandOperation[];

  @ApiProperty({ type: BackupOperation })
  @ValidateNested()
  @Field(() => BackupOperation)
  operation?: BackupOperation;

  @ApiProperty({ type: [ExecuteCommandOperation] })
  @ValidateNested()
  @Field(() => [ExecuteCommandOperation])
  postCommands?: ExecuteCommandOperation[];
}

/**
 * Config file for one Host
 *
 * Contains all information that can be used to backup a host.
 */
@ObjectType()
export class HostConfiguration {
  @ApiProperty({ type: Boolean })
  @Field(() => Boolean)
  isLocal?: boolean;

  @Exclude()
  password: string;

  @ApiProperty({ example: [] })
  addresses?: string[];

  @ValidateNested()
  dhcp?: DhcpAddress[];

  @ValidateNested()
  operations?: HostConfigOperation = new HostConfigOperation();

  @ValidateNested()
  schedule?: Schedule;
}
