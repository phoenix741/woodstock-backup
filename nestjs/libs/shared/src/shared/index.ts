export * from './backup.dto.js';
export * from './backups-files.dto.js';
export * from './host-configuration.dto.js';
export * from './scheduler.dto.js';
export * from './stats.dto.js';
export * from './tasks.dto.js';
export * from './woodstock-client.interface.js';
export * from './woodstock.constant.js';
export * from './woodstock.interface.js';
export * from './woodstock.model.js';
