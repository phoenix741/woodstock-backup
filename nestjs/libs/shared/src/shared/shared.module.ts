import { Module } from '@nestjs/common';
import { ApplicationConfigModule } from '../config';
import { LockService } from './lock.service';

@Module({
  imports: [ApplicationConfigModule],
  providers: [LockService],
  exports: [LockService],
})
export class SharedModule {}
