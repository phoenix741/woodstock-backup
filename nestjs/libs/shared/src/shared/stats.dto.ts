import { Field, Int, ObjectType } from '@nestjs/graphql';
import { Transform } from 'class-transformer';

@ObjectType()
export class BigIntTimeSerie {
  time: number;

  @Transform((v) => v.value && BigInt(v.value))
  @Field(() => BigInt)
  value: bigint;

  constructor(o: Partial<BigIntTimeSerie>) {
    Object.assign(this, o);
  }
}

@ObjectType()
export class NumberTimeSerie {
  time: number;

  @Field(() => Int)
  value: number;

  constructor(o: NumberTimeSerie) {
    Object.assign(this, o);
  }
}

@ObjectType()
export class DiskUsage {
  @Transform((v) => v.value && BigInt(v.value))
  @Field(() => BigInt)
  used?: bigint;
  usedRange?: BigIntTimeSerie[];
  @Transform((v) => v.value && BigInt(v.value))
  @Field(() => BigInt)
  usedLastMonth?: bigint;

  @Transform((v) => v.value && BigInt(v.value))
  @Field(() => BigInt)
  free?: bigint;
  freeRange?: BigIntTimeSerie[];
  @Transform((v) => v.value && BigInt(v.value))
  @Field(() => BigInt)
  freeLastMonth?: bigint;

  @Transform((v) => v.value && BigInt(v.value))
  @Field(() => BigInt)
  total?: bigint;
  totalRange?: BigIntTimeSerie[];
  @Transform((v) => v.value && BigInt(v.value))
  @Field(() => BigInt)
  totalLastMonth?: bigint;
}

@ObjectType()
export class PoolUsage {
  @Field(() => Int)
  longestChain?: number;
  longestChainRange?: NumberTimeSerie[];
  @Field(() => Int)
  longestChainLastMonth?: number;

  @Field(() => Int)
  nbChunk?: number;
  nbChunkRange?: NumberTimeSerie[];
  @Field(() => Int)
  nbChunkLastMonth?: number;

  @Field(() => Int)
  nbRef?: number;
  nbRefRange?: NumberTimeSerie[];
  @Field(() => Int)
  nbRefLastMonth?: number;

  @Transform((v) => v.value && BigInt(v.value))
  @Field(() => BigInt)
  size?: bigint;
  sizeRange?: BigIntTimeSerie[];
  @Transform((v) => v.value && BigInt(v.value))
  @Field(() => BigInt)
  sizeLastMonth?: bigint;

  @Transform((v) => v.value && BigInt(v.value))
  @Field(() => BigInt)
  compressedSize?: bigint;
  compressedSizeRange?: BigIntTimeSerie[];
  @Transform((v) => v.value && BigInt(v.value))
  @Field(() => BigInt)
  compressedSizeLastMonth?: bigint;

  @Transform((v) => v.value && BigInt(v.value))
  @Field(() => BigInt)
  unusedSize?: bigint;
  unusedSizeRange?: BigIntTimeSerie[];
  @Transform((v) => v.value && BigInt(v.value))
  @Field(() => BigInt)
  unusedSizeLastMonth?: bigint;
}

@ObjectType()
export class HostStatistics {
  host?: string;

  @Field(() => Int)
  longestChain?: number;
  longestChainRange?: NumberTimeSerie[];
  @Field(() => Int)
  longestChainLastMonth?: number;

  @Field(() => Int)
  nbChunk?: number;
  nbChunkRange?: NumberTimeSerie[];
  @Field(() => Int)
  nbChunkLastMonth?: number;

  @Field(() => Int)
  nbRef?: number;
  nbRefRange?: NumberTimeSerie[];
  @Field(() => Int)
  nbRefLastMonth?: number;

  @Transform((v) => v.value && BigInt(v.value))
  @Field(() => BigInt)
  size?: bigint;
  sizeRange?: BigIntTimeSerie[];
  @Transform((v) => v.value && BigInt(v.value))
  @Field(() => BigInt)
  sizeLastMonth?: bigint;

  @Transform((v) => v.value && BigInt(v.value))
  @Field(() => BigInt)
  compressedSize?: bigint;
  compressedSizeRange?: BigIntTimeSerie[];
  @Transform((v) => v.value && BigInt(v.value))
  @Field(() => BigInt)
  compressedSizeLastMonth?: bigint;
}

@ObjectType()
export class Statistics {
  diskUsage?: DiskUsage;
  poolUsage?: PoolUsage;
  hosts?: HostStatistics[];
}
