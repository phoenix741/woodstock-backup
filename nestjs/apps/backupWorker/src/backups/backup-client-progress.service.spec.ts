import { Logger } from '@nestjs/common';
import { ClientGrpcProxy } from '@nestjs/microservices';
import { Test, TestingModule } from '@nestjs/testing';
import { bigIntToLong, EntryType } from '@woodstock/shared';
import { from, lastValueFrom, toArray } from 'rxjs';
import { BackupsGrpcContext } from './backup-client-grpc.class.js';
import { BackupClientProgress } from './backup-client-progress.service.js';
import { BackupClient } from './backup-client.service.js';

describe('BackupClientProgress', () => {
  const mockBackupClient = {
    authenticate: () => 0,
    executeCommand: () => 0,
    getFileList: () => 0,
    createBackup: () => 0,
    refreshCache: () => 0,
    countRef: () => 0,
    close: () => 0,
  };

  const fakeClient = {} as ClientGrpcProxy;

  let backupClientProgress: BackupClientProgress;

  function createChunks(size: bigint) {
    const SPLIT = 75n;
    const chunks = [];
    for (let i = 0n; i < size / SPLIT; i++) {
      chunks.push({
        sha256: Buffer.from('sha256_' + i),
        size: SPLIT,
        compressedSize: 50n,
      });
    }
    chunks.push({
      sha256: Buffer.from('sha256_' + size / SPLIT),
      size: size % SPLIT,
      compressedSize: 50n,
    });
    return chunks;
  }

  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      providers: [{ provide: BackupClient, useValue: mockBackupClient }, BackupClientProgress],
    }).compile();

    backupClientProgress = module.get<BackupClientProgress>(BackupClientProgress);
  });

  it('#authenticate', async () => {
    // GIVEN
    mockBackupClient.authenticate = jest.fn().mockResolvedValue(undefined);

    const logger = new Logger('FakeLogger');
    const clientLogger = new Logger('FakeClientLogger');
    const ctxt = new BackupsGrpcContext('host', 'ip', 1, fakeClient);

    // WHEN
    const observable = backupClientProgress.authenticate(ctxt, logger, clientLogger, 'password');
    const result = await lastValueFrom(observable.pipe(toArray()));

    // THEN
    expect(result).toMatchSnapshot('result');
  });

  it('#executeCommand', async () => {
    // GIVEN
    mockBackupClient.executeCommand = jest.fn().mockResolvedValue(undefined);

    const ctxt = new BackupsGrpcContext('host', 'ip', 1, fakeClient);

    // WHEN
    const observable = backupClientProgress.executeCommand(ctxt, 'command');
    const result = await lastValueFrom(observable.pipe(toArray()));

    // THEN
    expect(result).toMatchSnapshot('result');
  });

  it('#getFileList', async () => {
    // GIVEN
    const it = [
      { type: EntryType.ADD, manifest: { path: Buffer.from('file1'), stats: { size: bigIntToLong(100n) } } },
      { type: EntryType.ADD, manifest: { path: Buffer.from('file2'), stats: { size: bigIntToLong(200n) } } },
      { type: EntryType.ADD, manifest: { path: Buffer.from('file3'), stats: { size: bigIntToLong(300n) } } },
      { type: EntryType.ADD, manifest: { path: Buffer.from('file4'), stats: { size: bigIntToLong(400n) } } },
      { type: EntryType.ADD, manifest: { path: Buffer.from('file5'), stats: { size: bigIntToLong(500n) } } },
      { type: EntryType.REMOVE, manifest: { path: Buffer.from('file6') } },
    ];
    mockBackupClient.getFileList = jest.fn().mockReturnValue(from(it));

    const ctxt = new BackupsGrpcContext('host', 'ip', 1, fakeClient);

    // WHEN
    const observable = backupClientProgress.getFileList(ctxt, {
      sharePath: Buffer.from('sharePath'),
      includes: [],
      excludes: [],
    });
    const result = await lastValueFrom(observable.pipe(toArray()));

    // THEN
    expect(result).toMatchSnapshot('result');
  });

  it('#createBackup', async () => {
    // GIVEN
    const it = [
      ...createChunks(100n),
      {
        type: EntryType.ADD,
        manifest: {
          path: Buffer.from('file1'),
          stats: { size: bigIntToLong(100n), compressedSize: bigIntToLong(50n) },
        },
      },
      ...createChunks(200n),
      {
        type: EntryType.ADD,
        manifest: {
          path: Buffer.from('file2'),
          stats: { size: bigIntToLong(200n), compressedSize: bigIntToLong(100n) },
        },
      },
      ...createChunks(300n),
      {
        type: EntryType.ADD,
        manifest: {
          path: Buffer.from('file3'),
          stats: { size: bigIntToLong(300n), compressedSize: bigIntToLong(150n) },
        },
      },
      ...createChunks(400n),
      {
        type: EntryType.ADD,
        manifest: {
          path: Buffer.from('file4'),
          stats: { size: bigIntToLong(400n), compressedSize: bigIntToLong(200n) },
        },
      },
      ...createChunks(500n),
      {
        type: EntryType.ADD,
        manifest: {
          path: Buffer.from('file5'),
          stats: { size: bigIntToLong(500n), compressedSize: bigIntToLong(250n) },
        },
      },
      ...createChunks(600n),
      {
        type: EntryType.ADD,
        manifest: {
          path: Buffer.from('file6'),
          stats: { size: bigIntToLong(600n), compressedSize: bigIntToLong(600n) },
        },
      },
    ];
    mockBackupClient.createBackup = jest.fn().mockReturnValue(from(it));

    const ctxt = new BackupsGrpcContext('host', 'ip', 1, fakeClient);

    // WHEN
    const observable = backupClientProgress.createBackup(ctxt, {
      sharePath: Buffer.from('sharePath'),
      includes: [],
      excludes: [],
    });
    const result = await lastValueFrom(observable.pipe(toArray()));

    // THEN
    for (const r of result) {
      expect(r).toMatchSnapshot({ speed: expect.any(Number) }, 'result');
    }
  });
});
