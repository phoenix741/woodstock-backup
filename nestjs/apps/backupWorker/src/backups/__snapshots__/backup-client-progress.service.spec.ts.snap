// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackupClientProgress #authenticate: result 1`] = `
[
  QueueTaskProgression {
    "compressedFileSize": 0n,
    "fileCount": 0,
    "fileSize": 0n,
    "newCompressedFileSize": 0n,
    "newFileCount": 0,
    "newFileSize": 0n,
    "progressCurrent": 0n,
    "progressMax": 0n,
    "speed": 0,
  },
  QueueTaskProgression {
    "compressedFileSize": 0n,
    "fileCount": 0,
    "fileSize": 0n,
    "newCompressedFileSize": 0n,
    "newFileCount": 0,
    "newFileSize": 0n,
    "progressCurrent": 0n,
    "progressMax": 0n,
    "speed": 0,
  },
  QueueTaskProgression {
    "compressedFileSize": 0n,
    "fileCount": 0,
    "fileSize": 0n,
    "newCompressedFileSize": 0n,
    "newFileCount": 0,
    "newFileSize": 0n,
    "progressCurrent": 1n,
    "progressMax": 1n,
    "speed": 0,
  },
]
`;

exports[`BackupClientProgress #createBackup: result 1`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 0n,
  "newFileCount": 0,
  "newFileSize": 0n,
  "progressCurrent": 0n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 2`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 0n,
  "newFileCount": 0,
  "newFileSize": 0n,
  "progressCurrent": 75n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 3`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 0n,
  "newFileCount": 0,
  "newFileSize": 0n,
  "progressCurrent": 100n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 4`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 50n,
  "newFileCount": 1,
  "newFileSize": 100n,
  "progressCurrent": 100n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 5`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 50n,
  "newFileCount": 1,
  "newFileSize": 100n,
  "progressCurrent": 175n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 6`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 50n,
  "newFileCount": 1,
  "newFileSize": 100n,
  "progressCurrent": 250n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 7`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 50n,
  "newFileCount": 1,
  "newFileSize": 100n,
  "progressCurrent": 300n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 8`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 150n,
  "newFileCount": 2,
  "newFileSize": 300n,
  "progressCurrent": 300n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 9`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 150n,
  "newFileCount": 2,
  "newFileSize": 300n,
  "progressCurrent": 375n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 10`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 150n,
  "newFileCount": 2,
  "newFileSize": 300n,
  "progressCurrent": 450n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 11`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 150n,
  "newFileCount": 2,
  "newFileSize": 300n,
  "progressCurrent": 525n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 12`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 150n,
  "newFileCount": 2,
  "newFileSize": 300n,
  "progressCurrent": 600n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 13`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 150n,
  "newFileCount": 2,
  "newFileSize": 300n,
  "progressCurrent": 600n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 14`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 300n,
  "newFileCount": 3,
  "newFileSize": 600n,
  "progressCurrent": 600n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 15`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 300n,
  "newFileCount": 3,
  "newFileSize": 600n,
  "progressCurrent": 675n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 16`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 300n,
  "newFileCount": 3,
  "newFileSize": 600n,
  "progressCurrent": 750n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 17`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 300n,
  "newFileCount": 3,
  "newFileSize": 600n,
  "progressCurrent": 825n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 18`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 300n,
  "newFileCount": 3,
  "newFileSize": 600n,
  "progressCurrent": 900n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 19`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 300n,
  "newFileCount": 3,
  "newFileSize": 600n,
  "progressCurrent": 975n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 20`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 300n,
  "newFileCount": 3,
  "newFileSize": 600n,
  "progressCurrent": 1000n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 21`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 500n,
  "newFileCount": 4,
  "newFileSize": 1000n,
  "progressCurrent": 1000n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 22`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 500n,
  "newFileCount": 4,
  "newFileSize": 1000n,
  "progressCurrent": 1075n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 23`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 500n,
  "newFileCount": 4,
  "newFileSize": 1000n,
  "progressCurrent": 1150n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 24`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 500n,
  "newFileCount": 4,
  "newFileSize": 1000n,
  "progressCurrent": 1225n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 25`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 500n,
  "newFileCount": 4,
  "newFileSize": 1000n,
  "progressCurrent": 1300n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 26`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 500n,
  "newFileCount": 4,
  "newFileSize": 1000n,
  "progressCurrent": 1375n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 27`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 500n,
  "newFileCount": 4,
  "newFileSize": 1000n,
  "progressCurrent": 1450n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 28`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 500n,
  "newFileCount": 4,
  "newFileSize": 1000n,
  "progressCurrent": 1500n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 29`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 750n,
  "newFileCount": 5,
  "newFileSize": 1500n,
  "progressCurrent": 1500n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 30`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 750n,
  "newFileCount": 5,
  "newFileSize": 1500n,
  "progressCurrent": 1575n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 31`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 750n,
  "newFileCount": 5,
  "newFileSize": 1500n,
  "progressCurrent": 1650n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 32`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 750n,
  "newFileCount": 5,
  "newFileSize": 1500n,
  "progressCurrent": 1725n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 33`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 750n,
  "newFileCount": 5,
  "newFileSize": 1500n,
  "progressCurrent": 1800n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 34`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 750n,
  "newFileCount": 5,
  "newFileSize": 1500n,
  "progressCurrent": 1875n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 35`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 750n,
  "newFileCount": 5,
  "newFileSize": 1500n,
  "progressCurrent": 1950n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 36`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 750n,
  "newFileCount": 5,
  "newFileSize": 1500n,
  "progressCurrent": 2025n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 37`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 750n,
  "newFileCount": 5,
  "newFileSize": 1500n,
  "progressCurrent": 2100n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 38`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 750n,
  "newFileCount": 5,
  "newFileSize": 1500n,
  "progressCurrent": 2100n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #createBackup: result 39`] = `
{
  "compressedFileSize": 0n,
  "fileCount": 0,
  "fileSize": 0n,
  "newCompressedFileSize": 1350n,
  "newFileCount": 6,
  "newFileSize": 2100n,
  "progressCurrent": 2100n,
  "progressMax": 0n,
  "speed": Any<Number>,
}
`;

exports[`BackupClientProgress #executeCommand: result 1`] = `
[
  QueueTaskProgression {
    "compressedFileSize": 0n,
    "fileCount": 0,
    "fileSize": 0n,
    "newCompressedFileSize": 0n,
    "newFileCount": 0,
    "newFileSize": 0n,
    "progressCurrent": 0n,
    "progressMax": 0n,
    "speed": 0,
  },
  QueueTaskProgression {
    "compressedFileSize": 0n,
    "fileCount": 0,
    "fileSize": 0n,
    "newCompressedFileSize": 0n,
    "newFileCount": 0,
    "newFileSize": 0n,
    "progressCurrent": 0n,
    "progressMax": 0n,
    "speed": 0,
  },
  QueueTaskProgression {
    "compressedFileSize": 0n,
    "fileCount": 0,
    "fileSize": 0n,
    "newCompressedFileSize": 0n,
    "newFileCount": 0,
    "newFileSize": 0n,
    "progressCurrent": 1n,
    "progressMax": 1n,
    "speed": 0,
  },
]
`;

exports[`BackupClientProgress #getFileList: result 1`] = `
[
  QueueTaskProgression {
    "compressedFileSize": 0n,
    "fileCount": 0,
    "fileSize": 0n,
    "newCompressedFileSize": 0n,
    "newFileCount": 0,
    "newFileSize": 0n,
    "progressCurrent": 0n,
    "progressMax": 100n,
    "speed": 0,
  },
  QueueTaskProgression {
    "compressedFileSize": 0n,
    "fileCount": 0,
    "fileSize": 0n,
    "newCompressedFileSize": 0n,
    "newFileCount": 0,
    "newFileSize": 0n,
    "progressCurrent": 0n,
    "progressMax": 300n,
    "speed": 0,
  },
  QueueTaskProgression {
    "compressedFileSize": 0n,
    "fileCount": 0,
    "fileSize": 0n,
    "newCompressedFileSize": 0n,
    "newFileCount": 0,
    "newFileSize": 0n,
    "progressCurrent": 0n,
    "progressMax": 600n,
    "speed": 0,
  },
  QueueTaskProgression {
    "compressedFileSize": 0n,
    "fileCount": 0,
    "fileSize": 0n,
    "newCompressedFileSize": 0n,
    "newFileCount": 0,
    "newFileSize": 0n,
    "progressCurrent": 0n,
    "progressMax": 1000n,
    "speed": 0,
  },
  QueueTaskProgression {
    "compressedFileSize": 0n,
    "fileCount": 0,
    "fileSize": 0n,
    "newCompressedFileSize": 0n,
    "newFileCount": 0,
    "newFileSize": 0n,
    "progressCurrent": 0n,
    "progressMax": 1500n,
    "speed": 0,
  },
  QueueTaskProgression {
    "compressedFileSize": 0n,
    "fileCount": 0,
    "fileSize": 0n,
    "newCompressedFileSize": 0n,
    "newFileCount": 0,
    "newFileSize": 0n,
    "progressCurrent": 0n,
    "progressMax": 1500n,
    "speed": 0,
  },
]
`;
